src/lib/statistics/chunk_statistics/counting_quotient_filter.hpp

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <vector>
+#include <string>
+
+#include "cqf2.hpp"
+#include "cqf4.hpp"
+#include "cqf8.hpp"
+#include "cqf16.hpp"
+#include "cqf32.hpp"
+#include "types.hpp"
+
+#include "storage/base_segment.hpp"
+#include "abstract_filter.hpp"
+
+
+
+
+namespace opossum {
+
+/* Counting Quotient Filters allow you to keep track of which values are present in a segment and how often. Filters
+work approximately. If a membership query yields a positive result, the value is probably present but there is
+a chance of a false positive. If the query delivers a negative result, the item is guaranteed to not be contained.
+In the same way, items can be over counted but not under counted.
+CQF can be configured with quotient size, which determines the number of slots, and the remainder size, which
+corresponds to the slot size. At this time, the remainder size must be 2, 4, 8, 16 or 32. */
+
+template <typename ElementType>
+class CountingQuotientFilter : AbstractFilter {
+ public:
+  CountingQuotientFilter(uint8_t quotient_size, uint8_t remainder_size);
+  virtual ~CountingQuotientFilter();
+  CountingQuotientFilter(const CountingQuotientFilter& filter) = delete;
+  CountingQuotientFilter(CountingQuotientFilter&& filter) = default;
+  CountingQuotientFilter& operator=(const CountingQuotientFilter& filter) = delete;
+  CountingQuotientFilter& operator=(CountingQuotientFilter&& filter) = default;
+
+  void insert(ElementType value, uint64_t count);

```

Comment:
Who uses this?

Reply:
Right now nobody.

Reply:
Do you expect this to change? If you think this makes sense in the future, shouldn't we default count to 1 and get rid of the second version of the method?